### DISCUSSION TITLE
[Appello fac-simile] Malfunzionamento CU

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/architettura-degli-elaboratori#87] [Altro (M-Z)]
![Esame](https://github.com/sapienzastudentsnetwork/archelab2223/assets/107944476/ffff6d31-2287-49b2-9236-0b21eebde6ab)
![CPU](https://github.com/sapienzastudentsnetwork/archelab2223/assets/107944476/654697af-2c41-4458-b7f8-ec9b38f88b3a)
---
### Soluzione del Prof
Le soluzioni fornite dal professore potrebbero essere incomplete o poco chiare, quindi sono benvenute soluzioni che le completano, oppure soluzioni alternative e/o con spiegazioni più approfondite: 
- [Soluzione.pdf](https://github.com/sapienzastudentsnetwork/archelab2223/files/11665710/AE.-.Appello.fac-simile.-.E3.-.Scheda.di.calcolo.-.Calcolo.pdf)
```
# Soluzione 1
add $s0, $zero, $zero       # $s0 assume valore 0
add $s1, $zero, $zero       # $s1 assume valore 0
nor $s1, $s1, $s1           # $s1 assume valore 0xFFFFFFFF
beq $s1, $s0, Exit          # In caso di guasto, il risultato di $s1-$s0 (cioè 0xFF…F) verrà scritto in $s0
Exit:

# Soluzione 2
add $s0, $zero, $zero       # $s0 assume valore 0
sw  $s0, 0xFFFFFFFF($zero)  # In caso di guasto, la somma di 0 + 0xFFFFFFFF verrà scritta in $s0
```

**Comment by CiottoloMaggico** [comment]
![immagine](https://github.com/sapienzastudentsnetwork/archelab2223/assets/79638284/79f491bc-da1b-4717-a753-ad2c9fdd09f6)
Questa è la soluzione con il percorso delle istruzioni all'interno del datapath
